frontend/app/(auth)/recover-password/__tests__/page.test.tsx

```diff
@@ -0,0 +1,109 @@
+import { fireEvent, render } from "@testing-library/react";
+import { describe, expect, it, vi } from "vitest";
+import RecoverPassword from "../page";
+const mockUsePathname = vi.fn(() => "/previous-page");
+const mockRedirect = vi.fn((url: string) => ({ url }));
+
+vi.mock("next/navigation", () => ({
+  redirect: (url: string) => mockRedirect(url),
+  usePathname: () => mockUsePathname(),
+}));
+
+const mockUseSupabase = vi.fn(() => ({
+  supabase: {
+    auth: {
+      updateUser: vi.fn(),
+    },
+  },
+  session: {
+    user: {},
+  },
+}));
+
+vi.mock("@/lib/context/SupabaseProvider", () => ({
+  useSupabase: () => mockUseSupabase(),
+}));
+
+const mockPublish = vi.fn();
+
+vi.mock("@/lib/hooks/useToast", () => ({
+  useToast: () => ({
+    publish: mockPublish,
+  }),
+}));
+
+const mockTrack = vi.fn();
+
+vi.mock("@/services/analytics/useEventTracking", () => ({
+  useEventTracking: () => ({
+    track: mockTrack,
+  }),
+}));
+
+describe("RecoverPassword component", () => {
+  it("should render the password update form", () => {
+    mockUseSupabase.mockReturnValue({
+      //@ts-expect-error doing this for testing purposes
+      session: { user: undefined },
+    });
+    const { getByTestId } = render(<RecoverPassword />);
+    const passwordField = getByTestId("password-field");
+    const updateButton = getByTestId("update-button");
+
+    expect(passwordField).toBeDefined();
+    expect(updateButton).toBeDefined();
+  });
+
+  it("should update the password and shows success toast", async () => {
+    mockUseSupabase.mockReturnValue({
+      supabase: {
+        auth: {
+          updateUser: vi.fn(() => ({
+            data: {},
+          })),
+        },
+      },
+      session: { user: {} },
+    });
+    const { getByTestId } = render(<RecoverPassword />);
+    const passwordField = getByTestId("password-field");
+    const updateButton = getByTestId("update-button");
+
+    fireEvent.change(passwordField, { target: { value: "new-password" } });
+    fireEvent.click(updateButton);
+
+    await expect(mockTrack).toHaveBeenCalledTimes(1);
+    await expect(mockTrack).toHaveBeenCalledWith("UPDATE_PASSWORD");
+    await expect(mockPublish).toHaveBeenCalledTimes(1);
+    await expect(mockPublish).toHaveBeenCalledWith({
+      variant: "success",
+      text: "Password updated successfully!",
+    });
+  });
```

Comment:
To ensure that the password has been changed, we should verify that the `updateUser` function has also been called here